mllib/src/main/scala/org/apache/spark/mllib/linalg/EigenValueDecomposition.scala

```diff
@@ -79,6 +79,10 @@ private[mllib] object EigenValueDecomposition {
     // Mode 1: A*x = lambda*x, A symmetric
     iparam(6) = 1
 
+    require(ncv * n.toLong <= Integer.MAX_VALUE, "Product of 2*k*n must be smaller than " +
```

Comment:
Okay, I think i figured out what you are saying. Take a look. The check on k may be overkill, since by definition it must be less than n. Therefore k*n is always greater than k*k. There could always be that weird scenario though. Thanks for the help.